src/main/java/com/jmb/springfactory/service/GenericServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.jmb.springfactory.service;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.jmb.springfactory.dao.GenericMongoDao;
+import com.jmb.springfactory.exceptions.NotFoundException;
+import com.jmb.springfactory.exceptions.PersistenceLayerException;
+import com.jmb.springfactory.exceptions.ServiceLayerException;
+import com.jmb.springfactory.model.bo.BusinessObjectBase;
+import com.jmb.springfactory.model.dto.BaseDto;
+import com.jmb.springfactory.model.entity.BaseEntity;
+
+public abstract class GenericServiceImpl<T extends BaseEntity, D extends BaseDto, B extends BusinessObjectBase,
+    ID extends Serializable> extends GenericTransformerServiceImpl<T, D, B> implements GenericService<D, ID> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)